FEATURES.md

```diff
@@ -0,0 +1,182 @@
+  creation.
+- **Child windows**: Windows can be created relative to the client area of other windows, and parent
+  windows can be disabled in favor of child windows.
+  ***//DISCUSS: SHOULD THIS BE SUPPORTED?***
+
+### System information
+|Feature      |Windows|MacOS |Linux x11|Linux Wayland|Android|iOS    |Emscripten|
+|------------ | ----- | ---- | ------- | ----------- | ----- | ----- | -------- |
+|Monitor list |✔️    |✔️    |✔️       |✔️          |**N/A**|**N/A**|**N/A**   |
+
+#### Details
+- **Monitor list**: Retrieve the list of monitors and their metada, including which one is primary is applicable
+
+### Input handling
+|Feature                                 |Windows|MacOS |Linux x11|Linux Wayland|Android|iOS    |Emscripten|
+|--------------------------------------- | ----- | ---- | ------- | ----------- | ----- | ----- | -------- |
+|Mouse events                            |✔️    |▢#63  |✔️       |✔️          |**N/A**|**N/A**|✔️       |
+|Mouse set location                      |✔️    |✔️    |✔️       |❓           |**N/A**|**N/A**|**N/A**  |
+|Cursor grab                             |✔️    |▢#165 |▢#242    |❌#306      |**N/A**|**N/A**|✔️       |
+|Cursor icon                             |✔️    |✔️    |✔️       |❌#306      |**N/A**|**N/A**|❌       |
+|Touch events                            |✔️    |❌    |✔️       |✔️          |✔️    |✔️     |✔️       |
+|Multitouch                              |❓     |❌    |✔️       |✔️          |❓     |❌     |❌       |
+|Keyboard events                         |✔️    |✔️    |✔️       |✔️          |❓     |❌     |✔️       |
+|Drag & Drop                             |✔️    |✔️    |✔️       |❌#306      |❌    |❌     |❌       |
+|Clipboard #162                          |❌    |❌    |❌       |❌          |❌    |❌     |❌       |
+|Raw Device Events                       |▢*#??*|▢*#??*|▢*#??*   |❌          |❌    |❌     |❌       |
+|Gamepad/Joystick events ***//DISCUSS*** |❌    |❌    |❌       |❌          |❌    |❌     |❌       |
+|Device movement events ***//DISCUSS***  |❓     |❓     |❓       |❓           |❌    |❌     |❌       |
+
+#### Details
+- **Mouse events**: Generating mouse events associated with pointer motion, click, and scrolling events.
+- **Mouse set location**: Forcibly changing the location of the pointer.
+- **Cursor grab**: Locking the cursor so it cannot exit the client area of a window.
+- **Cursor icon**: Changing the cursor icon, or hiding the cursor.
+- **Touch events**: Single-touch events.
+- **Multitouch**: Multi-touch events, including cancellation of a gesture.
+- **Keyboard events**: Properly processing keyboard events using the user-specified keymap and
+  translating keypresses into UTF-8 characters, handling dead keys and IMEs.
+- **Drag & Drop**: Dragging content into winit, detecting when content enters, drops, or if the drop is cancelled.
+  ***//DISCUSS: WINIT SUPPORTS FILE DROPS, BUT NOT TEXT OR IMAGE DROPS***
```

Comment:
Yes, this is a point I discussed in an earlier issue / PR. I don't know how this works for other platforms, but in wayland, drag'n'drop is unified for all kind of contents. The way it works is that the application starting the d'n'd and the application receiving it negociate a mime type for the data transfer. Dragging files from a file explorer is just a special case of this.

Can this kind of general behavior be transposed to other platforms?

Reply:
I assume? I know that it works on Windows, and I'm pretty sure I've done it before on macOS and X11 but I can't quite recall.

Reply:
I can confirm that it's similar on X11 [and macOS](https://github.com/tomaka/winit/pull/244#issuecomment-320134156), so I think we can reasonably provide this functionality.

Also, what about knowing the hover position and being able to decide whether to accept or reject a drop (based on both position and type)? Anyone building GUIs on top of winit will demand this, and it's at least easy to implement on X11.

Reply:
From skimming the Windows code, that also seems to be how Windows does it. The one question I have is how do we implement communication from the callback back to the operating system to accept or reject the drop? The best way I can think of doing it would be to pass a mutable reference along with the event that the callback can modify (as otherwise we're getting into the realm of having different return types for different events), but that isn't possible on Windows before Event Loop 2.0 gets merged.

Reply:
Another thing - Winit supports dragging content *into* windows. Should we support dragging content *from* windows? In principle I'm not opposed to it, but I don't know what the API would look like and it starts to turn Winit from *windowing and user input* to *general cross-platform desktop interfacing* which is a substantially bigger project.

Perhaps we could publicly expose some of Winit's internals in a permanently unstable way (say, hide them behind a feature or make them all `#[doc(hidden)]`) and split that into a separate crate, alongside non-windowing stuff like clipboard handling? Doing that would have issues and I'm honestly not entirely sold on it myself, but it would allow us to keep the core of Winit simple and build more specialized libraries around it.